transformers/pipelines.py

```diff
@@ -0,0 +1,815 @@
+        }
+    },
+    'question-answering': {
+        'impl': QuestionAnsweringPipeline,
+        'tf': TFAutoModelForQuestionAnswering if is_tf_available() else None,
+        'pt': AutoModelForQuestionAnswering if is_torch_available() else None,
+        'default': {
+            'model': 'distilbert-base-uncased-distilled-squad',
+            'config': None,
+            'tokenizer': 'bert-base-uncased'
+        }
+    }
+}
+
+
+def pipeline(task: str, model: Optional = None,
+             config: Optional[Union[str, PretrainedConfig]] = None,
+             tokenizer: Optional[Union[str, PreTrainedTokenizer]] = None, **kwargs) -> Pipeline:
+    """
+    Utility factory method to build a pipeline.
+    Pipeline are made of:
+        A Tokenizer instance in charge of mapping raw textual input to token
+        A Model instance
+        Some (optional) post processing for enhancing model's output
+
+    Examples:
+        pipeline('sentiment-analysis')
+        pipeline('question-answering', model='distilbert-base-uncased-distilled-squad', tokenizer='bert-base-cased')
+        pipeline('ner', model=AutoModel.from_pretrained(...), tokenizer=AutoTokenizer.from_pretrained(...)
+        pipeline('ner', model='https://...pytorch-model.bin', config='https://...config.json', tokenizer='bert-base-cased')
+    """
+    # Try to infer tokenizer from model name (if provided as str)
+    if tokenizer is None:
+        if model is not None and not isinstance(model, str):
+            # Impossible to guest what is the right tokenizer here
+            raise Exception('Tokenizer cannot be None if provided model is a PreTrainedModel instance')
+        else:
+            tokenizer = model
+
+    # Retrieve the task
+    if task not in SUPPORTED_TASKS:
+        raise KeyError("Unknown task {}, available tasks are {}".format(task, list(SUPPORTED_TASKS.keys())))
+
+    targeted_task = SUPPORTED_TASKS[task]
+    task, allocator = targeted_task['impl'], targeted_task['tf'] if is_tf_available() else targeted_task['pt']
+
+    # Handling for default model for the task
+    if model is None:
+        model, config, tokenizer = tuple(targeted_task['default'].values())
+
+    # Allocate tokenizer
+    tokenizer = tokenizer if isinstance(tokenizer, PreTrainedTokenizer) else AutoTokenizer.from_pretrained(tokenizer)
+
+    # Special handling for model conversion
+    if isinstance(model, str):
+        from_tf = model.endswith('.h5') and not is_tf_available()
+        from_pt = model.endswith('.bin') and not is_torch_available()
+
+        if from_tf:
+            logger.warning('Model might be a TensorFlow model (ending with `.h5`) but TensorFlow is not available. '
+                           'Trying to load the model with PyTorch.')
+        elif from_pt:
+            logger.warning('Model might be a PyTorch model (ending with `.bin`) but PyTorch is not available. '
+                           'Trying to load the model with Tensorflow.')
```

Comment:
From my understanding, loading the pytorch model simply won't work at all in this case since `torch` needs to be installed to load pytorch checkpoints into tf models, am I wrong? (same in the other way)